tests/unittests/Makefile

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) Microsoft. All rights reserved
+#
+
+include $(CLEAR_VARS)
+
+CLANG := clang++
+
+SRCDIR := Foundation
+SRCFILES := $(wildcard $(SRCDIR)/*.m*)
+RF_SRCFILES := $(wildcard $(SRCDIR)/ReferenceFoundation/*.mm)
+
+OUTDIR := ../../build/Tests/UnitTests/Foundation/OSX
+OUTTARGETS := $(addsuffix .o, $(addprefix $(OUTDIR)/, $(basename $(notdir $(SRCFILES)))))
+OUTTARGETS += $(addsuffix .o, $(addprefix $(OUTDIR)/ReferenceFoundation_, $(basename $(notdir $(RF_SRCFILES)))))
+
+SRC := Foundation/NSArrayTests.mm
+
+INC := ./Foundation/ReferenceFoundation ../frameworks/include ../frameworks/gtest ../frameworks/gtest/include 
+INC += ../frameworks/OSXShims/include Foundation/ReferenceFoundation ../../frameworks/include
+INC_PARAMS := $(INC:%=-I %)
+
+CFLAGS += -std=c++14
+CFLAGS += -x objective-c++
+CFLAGS += $(INC_PARAMS)
+
+Shims : $(OUTDIR)/windows.o $(OUTDIR)/IwMalloc.o
+
+$(OUTDIR)/IwMalloc.o : ../../Frameworks/WinObjCRT/MemoryManagement.cpp
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $<
+
+$(OUTDIR)/windows.o : ../frameworks/OSXShims/src/windows.cpp
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $<
+
+$(OUTDIR)/%.o : Shims $(SRCDIR)/%.m
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $(word 2,$^)
+
+$(OUTDIR)/%.o : Shims $(SRCDIR)/%.mm
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $(word 2,$^) 
+
+$(OUTDIR)/ReferenceFoundation_TestUtils.o : $(SRCDIR)/ReferenceFoundation/TestUtils.mm
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $<
+
+$(OUTDIR)/ReferenceFoundation_%.o : $(OUTDIR)/ReferenceFoundation_TestUtils.o $(SRCDIR)/ReferenceFoundation/%.mm
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $(word 2,$^) 
+
+.PHONY: clean
+
+clean:
+	rm $(OUTDIR)/*.o
+	rm Foundation.UnitTests.exe
+
+Foundation.UnitTests.exe : $(OUTTARGETS)
+	$(CLANG) -o $@ -std=c++14 $(INC_PARAMS) EntryPoint.cpp ../frameworks/gtest/src/gtest-all.cc -framework Foundation ../../build/Tests/UnitTests/Foundation/OSX/*.o
```

Comment:
removing -std=c++14 generates some concerning warnings. LDFLAGS added

---
In reply to: [70349983](https://github.com/Microsoft/WinObjC/pull/634#discussion_r70349983) [](ancestors = 70349983)